adal/oauth2_client.py

```diff
@@ -138,7 +138,7 @@ def _parse_id_token(self, encoded_token):
                 self._log.warn('The returned id_token could not be base64 url safe decoded.')
                 return
 
-            id_token = json.loads(b64_decoded.decode())
+            id_token = json.loads(b64_decoded.decode('utf-8'))

```

Comment:
In short, the current fix is sufficient for this particular case. But I would suggest we change at least one more line at line 145.

So this symptom can be demonstrated by the following test snippet running on Python 2:

    >>> utf8_version = u'Guinebertière'.encode('utf-8')
    >>> utf8_version.decode()  # In Python 2, the default encoding is 'ascii'
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 10: ordinal not in range(128)

At a hindsight, I would say [line 145](https://github.com/yugangw-msft/azure-activedirectory-library-for-python/blob/release.0.4.1/adal/oauth2_client.py#L145) should better NOT shadow the raw exception by a `return`. It should just use `raise`. That way it would give you more precise error trace than you've currently got in [that CLI issue](https://github.com/Azure/azure-cli/issues/1086), and probably easier to pinpoint the culprit.